lib/spack/spack/operating_systems/mac_os.py

```diff
@@ -18,6 +19,17 @@ def macos_version():
     return Version(py_platform.mac_ver()[0])
 
 
+def macos_cltools_version():
+    pkgutil = Executable('pkgutil')
+    output = pkgutil('--pkg-info=com.apple.pkg.CLTools_Executables',
+                     output=str, error=str, fail_on_error=False)
+    match = re.search(r'version:\s*([0-9.]+)', output)
+    if match:
+        return Version(match.group(1))
+
+    return None
+
+
```

Comment:
I'm a bit concerned here, not sure why but on my current machine (with the tools installed) this is what I get:

```
❯ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
No receipt for 'com.apple.pkg.CLTools_Executables' found at '/'.
```

Perhaps add an alternate regex for xcodebuild -version?  For whatever reason that one works.

Reply:
Interesting, thanks for the datapoint... this was fallout from discussion with @alalazo on the perl utils, #26290. It's not currently used, since the deployment target seems to fix the use case, but I thought it would be good to move into `mac_os` since it seems to be one of the many factors (compiler, xcode verison, clutils version, default sdk, deployment target) that affects macOS builds.

Reply:
On my computer (XCode 13, big sur, *not* downloaded the "Command Line Tools for Xcode 13.2" update):
```
$ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
package-id: com.apple.pkg.CLTools_Executables
version: 13.0.0.0.1.1630607135
volume: /
location: /
install-time: 1633264385
groups: com.apple.FindSystemFiles.pkg-group
$ pkgutil --pkg-info=com.apple.pkg.Xcode
package-id: com.apple.pkg.Xcode
version: 13.0.0.0.1.1630607135
volume: /
location: /
install-time: 1633262083
```

On my coworker who's presumably downloaded new xcode but not the latest clutils:
```
% pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
package-id: com.apple.pkg.CLTools_Executables
version: 13.2.0.0.1.1638488800
volume: /
location: /
install-time: 1643225472
groups: com.apple.FindSystemFiles.pkg-group
% pkgutil --pkg-info=com.apple.pkg.Xcode
package-id: com.apple.pkg.Xcode
version: 13.0.0.0.1.1630607135
volume: /
location: /
install-time: 1635167450
```

With XCode installed by downloading from apple and expanding into ~/Applications, the xcode package doesn't show up:
```
ornlci05:~ b1n$ pkgutil --pkg-info=com.apple.pkg.CLTools_Executables
package-id: com.apple.pkg.CLTools_Executables
version: 13.2.0.0.1.1638488800
volume: /
location: /
install-time: 1644416245
groups: com.apple.FindSystemFiles.pkg-group 
ornlci05:~ b1n$ pkgutil --pkg-info=com.apple.pkg.Xcode
No receipt for 'com.apple.pkg.Xcode' found at '/'.
```